x-pack/plugins/apm/server/lib/services/get_services/get_services_items.ts

```diff
@@ -19,72 +20,315 @@ import {
 } from '../../helpers/setup_request';
 import { getServicesProjection } from '../../../../common/projections/services';
 
+const MAX_NUMBER_OF_SERVICES = 500;
+
 export type ServiceListAPIResponse = PromiseReturnType<typeof getServicesItems>;
-export async function getServicesItems(
-  setup: Setup & SetupTimeRange & SetupUIFilters
-) {
-  const { start, end, client } = setup;
 
-  const projection = getServicesProjection({ setup });
+const arrayUnionToCallable = <T extends any[]>(
+  array: T
+): Array<ValuesType<T>> => {
+  return array;
+};
+
+type ServicesItemsSetup = Setup & SetupTimeRange & SetupUIFilters;
+
+type ServicesItemsProjection = ReturnType<typeof getServicesProjection>;
+
+interface AggregationParams {
+  setup: ServicesItemsSetup;
+  projection: ServicesItemsProjection;
+}
 
-  const params = mergeProjection(projection, {
-    body: {
+const getTransactionDurationAvg = async ({
+  setup,
+  projection,
+}: AggregationParams) => {
+  const { client } = setup;
+
+  const response = await client.search(
+    mergeProjection(projection, {
       size: 0,
-      aggs: {
-        services: {
-          terms: {
-            ...projection.body.aggs.services.terms,
-            size: 500,
+      body: {
+        query: {
+          bool: {
+            filter: projection.body.query.bool.filter.concat({
+              term: {
+                [PROCESSOR_EVENT]: 'transaction',
+              },
+            }),
           },
-          aggs: {
-            avg: {
-              avg: { field: TRANSACTION_DURATION },
+        },
+        aggs: {
+          services: {
+            terms: {
+              ...projection.body.aggs.services.terms,
+              size: MAX_NUMBER_OF_SERVICES,
             },
-            agents: {
-              terms: { field: AGENT_NAME, size: 1 },
+            aggs: {
+              average: {
+                avg: {
+                  field: TRANSACTION_DURATION,
+                },
+              },
+            },
+          },
+        },
+      },
+    })
+  );
+
+  const { aggregations } = response;
+
+  if (!aggregations) {
+    return [];
+  }
+
+  return aggregations.services.buckets.map((bucket) => ({
+    name: bucket.key as string,
+    value: bucket.average.value,
+  }));
+};
+
+const getAgentName = async ({ setup, projection }: AggregationParams) => {
+  const response = await setup.client.search(
+    mergeProjection(projection, {
+      body: {
+        query: {
+          bool: {
+            filter: [
+              ...projection.body.query.bool.filter,
```

Comment:
In the `getTransactionDurationAvg` you used:

```js
filter: projection.body.query.bool.filter.concat
```

and here you used spread syntax. 
```js
...projection.body.query.bool.filter,
```
I'm not against neither implementation, but maybe we could pick one so we have consistency?